Minichain_v0/state.py

```
@@ -0,0 +1,70 @@
+    # Validate signature
+    if not tx.verify():
+        raise ValueError("Invalid signature")
+    
+    # Validate nonce (prevents replay attacks)
+    expected_nonce = state.get_nonce(tx.sender)
+    if tx.nonce != expected_nonce:
+        raise ValueError(f"Bad nonce: expected {expected_nonce}, got {tx.nonce}")
+    
+    # Validate balance
+    if state.get_balance(tx.sender) < tx.amount:
+        raise ValueError(f"Insufficient balance")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)